### ISSUE TITLE
Circle CI "userns remapping failure"

### ISSUE THREAD

**Issue by arashgm** [cirruslabs/docker-images-flutter#95]
we are getting error on Circle CI on this container that just pushed : cirrusci/flutter:1.20.3

CircleCI was unable to start the container because of a userns remapping failure in Docker.
would you please take a look

**Comment by komarekw** [comment]
here you can finde a bit more details:
`CircleCI was unable to start the container because of a userns remapping failure in Docker.

This typically means the image contains files with UID/GID values that are higher than Docker and CircleCI can use.

Checkout our docs https://circleci.com/docs/2.0/high-uid-error/ to learn how to fix this problem.
Original error: CircleCI was unable to start the container because of a userns remapping failure in Docker.

This typically means the image contains files with UID/GID values that are higher than Docker and CircleCI can use.

Checkout our docs https://circleci.com/docs/2.0/high-uid-error/ to learn how to fix this problem.
Original error: failed to register layer: Error processing tar file(exit status 1): Container ID 397546 cannot be mapped to a host ID`

**Comment by fkorotkov** [comment]
@arashgm @komarekw could you please provide your Circle config? At least the part where you use the image. This is definitely caused by #93 which fixed #45.

**Comment by arashgm** [comment]
@fkorotkov 

```
docker:
      - image: cirrusci/flutter:1.20.3
    environment:
      TERM: dumb
      _JAVA_OPTIONS: "-Xmx2048m -XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap"
      GRADLE_OPTS: '-Dorg.gradle.jvmargs="-Xmx2048m"'
```

**Comment by fkorotkov** [comment]
Hm, `find / \( -uid 1000000 \)  -ls 2>/dev/null` from the [documentation](https://circleci.com/docs/2.0/high-uid-error/#solution) doesn't show anything.

**Comment by komarekw** [comment]
can you search for -uid 397546?

**Comment by fkorotkov** [comment]
All users are under 65535:

```bash
$ docker run -it cirrusci/flutter:latest sh
# cat /etc/passwd       
root:x:0:0:root:/root:/bin/bash
daemon:x:1:1:daemon:/usr/sbin:/usr/sbin/nologin
bin:x:2:2:bin:/bin:/usr/sbin/nologin
sys:x:3:3:sys:/dev:/usr/sbin/nologin
sync:x:4:65534:sync:/bin:/bin/sync
games:x:5:60:games:/usr/games:/usr/sbin/nologin
man:x:6:12:man:/var/cache/man:/usr/sbin/nologin
lp:x:7:7:lp:/var/spool/lpd:/usr/sbin/nologin
mail:x:8:8:mail:/var/mail:/usr/sbin/nologin
news:x:9:9:news:/var/spool/news:/usr/sbin/nologin
uucp:x:10:10:uucp:/var/spool/uucp:/usr/sbin/nologin
proxy:x:13:13:proxy:/bin:/usr/sbin/nologin
www-data:x:33:33:www-data:/var/www:/usr/sbin/nologin
backup:x:34:34:backup:/var/backups:/usr/sbin/nologin
list:x:38:38:Mailing List Manager:/var/list:/usr/sbin/nologin
irc:x:39:39:ircd:/var/run/ircd:/usr/sbin/nologin
gnats:x:41:41:Gnats Bug-Reporting System (admin):/var/lib/gnats:/usr/sbin/nologin
nobody:x:65534:65534:nobody:/nonexistent:/usr/sbin/nologin
_apt:x:100:65534::/nonexistent:/usr/sbin/nologin
messagebus:x:101:101::/nonexistent:/usr/sbin/nologin
systemd-timesync:x:102:102:systemd Time Synchronization,,,:/run/systemd:/usr/sbin/nologin
systemd-network:x:103:104:systemd Network Management,,,:/run/systemd:/usr/sbin/nologin
systemd-resolve:x:104:105:systemd Resolver,,,:/run/systemd:/usr/sbin/nologin
```

**Comment by fkorotkov** [comment]
Explicitly setting user to `root` and trying to `chown -R root:root` didn't help.

**Comment by komarekw** [comment]
I guess it is after downloading flutter SDK.

**Comment by fkorotkov** [comment]
I'm running out of ideas. Pinged [Circle CI Support on Twitter](https://twitter.com/fedor/status/1305536235415445504). In the meantime setting the image to [the latest `sha256`](https://cirrus-ci.com/task/5814875731525632?command=push#L48) before the breaking changes in #93 mitigates the issue:

```yaml
docker:
      - image: cirrusci/flutter@sha256:f124cbbe0ea2caa53135db8d7f4e427f8ad091cb09e0202637c1547f07ad7b1b
```

**Comment by SemanticallyNull** [comment]
It looks like the grade wrapper scripts in the container have UIDs that are above 65535. See:

```
root@562040db3bf6:/sdks/flutter# ls -lan /sdks/flutter/bin/cache/artifacts/gradle_wrapper/gradle
[...]
drwxr-xr-x 2 397546 5000 4096 Sep 14 16:01 wrapper
```

Here's the list of files:
* `/sdks/flutter/bin/cache/artifacts/gradle_wrapper/gradlew.bat
* `/sdks/flutter/bin/cache/artifacts/gradle_wrapper/gradle`
* `/sdks/flutter/bin/cache/artifacts/gradle_wrapper/gradle/wrapper`
* `/sdks/flutter/bin/cache/artifacts/gradle_wrapper/gradle/wrapper/gradle-wrapper.jar`
* `/sdks/flutter/bin/cache/artifacts/gradle_wrapper/gradlew`

**Comment by fkorotkov** [comment]
Thank you, @SemanticallyNull! 96b86c161bbf204ffd11f8590e08726d796e8cb1 should fix it shortly then. 🎉

**Comment by fkorotkov** [comment]
Can confirm, it's working now! Thank you again, @SemanticallyNull! 🙌

**Comment by komarekw** [comment]
is this also merged to 1.20.3? Because we use this version.

**Comment by ened** [comment]
@fkorotkov this may have broken the "wait_for_emulator" task over in https://github.com/BlueChilli/flutter_uploader/blob/79c641b66ee10b905ec33f921b6f892488df1d12/.cirrus.yml. Used to be fine with 1.20.2, not stopped working with 1.22.0.
